Microsoft.Toolkit.Win32/Microsoft.Toolkit.Win32.UI.Controls/Interop/WinRT/UriToLocalStreamResolver.cs

```diff
@@ -1,63 +1,87 @@
-// Licensed to the .NET Foundation under one or more agreements.
+﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
 using System;
 using System.IO;
 using System.Runtime.InteropServices;
-using System.Threading.Tasks;
 using Microsoft.Toolkit.Win32.UI.Controls.Interop.Win32;
-using Windows.Foundation;
-using Windows.Storage.Streams;
-using Windows.Web;
 
 namespace Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT
 {
-    internal sealed class UriToLocalStreamResolver : IUriToStreamResolver
+    internal class UriToLocalStreamResolver : IUriToStreamResolver
     {
-        public IAsyncOperation<IInputStream> UriToStreamAsync(Uri uri)
+        private readonly string _path;
+
+        public UriToLocalStreamResolver()
+#pragma warning disable SA1129 // Do not use default value type constructor
+            : this(Path.GetDirectoryName(UnsafeNativeMethods.GetModuleFileName(new HandleRef())))
+#pragma warning restore SA1129 // Do not use default value type constructor
+        {
+        }
+
+        protected UriToLocalStreamResolver(string path)
+        {
+            _path = path;
+        }
+
+        public Stream UriToStream(Uri uri)
+        {
+            var fullPath = ConvertToPath(uri);
+
+            // TODO: Make this proper async all the way through
```

Comment:
Is there an open work item for this?

Reply:
In 42f2916ed01059778cc6e576568b3371bc4de928 I've marked the APIs that use this as obsolete, as it is replaced with a new API introduced in this PR. The class will remain, but only for tests, so it does not need to be async.